### ISSUE TITLE
Generate prom API from Swagger/OpenAPI

### ISSUE THREAD

**Issue by bwplotka** [prometheus/client_golang#896]
With the work done with @Hangzhi on https://github.com/thanos-io/thanos/issues/4102 I think we could replace current promclient with generated client or remove it completely (v2)

**Comment by PhilipGough** [comment]
@bwplotka have you had any success with generating relatively idiomatic clients from OpenAPI spec? If so, what tools did you use?

**Comment by bwplotka** [comment]
No. I did not XD but did not try much... this has to be experimented with

**Comment by PhilipGough** [comment]
@saswatamcode - this may be of interest to you since you have done something similar in https://github.com/observatorium/obsctl/ recently I think :)

**Comment by bwplotka** [comment]
This has been done in https://prometheus.io/docs/prometheus/latest/querying/api/#openapi-specification
